src/PhenX.EntityFrameworkCore.BulkInsert.Oracle/OracleBulkInsertProvider.cs

```diff
@@ -0,0 +1,90 @@
+using JetBrains.Annotations;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+using Oracle.ManagedDataAccess.Client;
+
+using PhenX.EntityFrameworkCore.BulkInsert.Metadata;
+using PhenX.EntityFrameworkCore.BulkInsert.Options;
+
+namespace PhenX.EntityFrameworkCore.BulkInsert.Oracle;
+
+[UsedImplicitly]
+internal class OracleBulkInsertProvider(ILogger<OracleBulkInsertProvider>? logger) : BulkInsertProviderBase<OracleDialectBuilder, OracleBulkInsertOptions>(logger)
+{
+    /// <inheritdoc />
+    protected override string BulkInsertId => "ROWID";
+
+    /// <inheritdoc />
+    protected override string AddTableCopyBulkInsertId => ""; // No need to add an ID column in Oracle
+
+    /// <inheritdoc />
+    protected override string GetTempTableName(string tableName) => $"#temp_bulk_insert_{Guid.NewGuid().ToString("N")[..8]}";
```

Comment:
Why do you not take the whole guid?

Reply:
oracle under a certain version (12 I think) does not allow indentifiers longer than 30, yay

Reply:
I would probably comment that, because I would have forgotten that next week.

Reply:
oh I will remember 😅 but yeah I'll add a comment